drivers/android/transaction.rs

```diff
@@ -16,7 +26,12 @@ use crate::{
     DeliverToRead, Either,
 };
 
+struct TransactionInner {
+    file_list: List<Box<FileInfo>>,
+}
+
 pub(crate) struct Transaction {
+    inner: SpinLock<TransactionInner>,
```

Comment:
Do you have a real need for a `SpinLock` here, as opposed to a `Mutex`? Are you locking this from a non-sleep context?
(Not suggesting it should be a Mutex - I just genuinely don't know)

Reply:
I need some synchronisation. Given that the hold times are small (just to swap a pointer), I chose a spinlock because it will just spin on contention as opposed to sleep. This is preferable because `schedule()` itself would take longer than the hold time of the lock, that is, sleeping is counterproductive on contention.

Reply:
> just to swap a pointer

Would `AtomicPtr` work?

Reply:
> > just to swap a pointer
> 
> Would `AtomicPtr` work?

I considered it, but decided against it for two reasons: it doesn't work for pointers to `?Sized` types, and I may want to add more fields to it `inner` eventually.

Reply:
Makes sense. Could be one of the parameters to be verified using the ping benchmark somewhere in the future?

Reply:
Yes, absolutely.

One "advantage" we have with binder is that we have the C implementation to benchmark against. As long as we're not too far off, we're good.